interview-questions.md

```diff
@@ -6,52 +6,416 @@ First, without external resources. Challenge yourself to answer from memory as i
+
+Functions can be declared as standalone entities, like this:
+
+        function myFunction(param1, param2) {
+            // function code here
+        }
+
+Functions can also be assigned to variables or passed as arguments to other functions, like this:
+
+        let myFunction = function(param1, param2) {
+            // function code here
+        }
+
+In contrast, methods are functions that are attached to objects and are called using dot notation, like this:
+
+
+        let myObject = {
+            myMethod: function() {
+            // method code here
+            }
+        }
+
+        myObject.myMethod(); // call the method on the object
+
+In this example, myMethod is a method that is attached to the myObject object. The method can be called using dot notation on the object.
+
+Source: 
+
+MDN Web Docs: Functions - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Functions
+
+MDN Web Docs: Working with objects - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Working_with_Objects
+
+-------
+
+### 6. STRETCH: What is hoisting in JavaScript?
 
 Your answer:
 
+ Hoisting is a JavaScript mechanism where variable and function declarations are moved to the top of their respective scopes before code execution.
+
 Researched answer:
 
+Hoisting is a mechanism in JavaScript where variable and function declarations are moved to the top of their respective scopes before code execution. This means that variables and functions can be declared after they are used in code. However, only the declaration is moved, not the initialization or assignment.  
+
+For example, the following code will still work even though the function is called before it is declared:  
+
+    
+        myFunction(); // "Hello, world!"
+
+        function myFunction() {
+            console.log("Hello, world!")
+        }
+
+In this example, the myFunction function is called before it is declared, but the code still works because the function declaration is hoisted to the top of the scope.  
+
+Variables are also hoisted in JavaScript, but with a subtle difference. Variable declarations are hoisted, but not their assignments. This means that variables can be declared after they are used in code, but they will be undefined until they are assigned a value.
+
+For example, the following code will log undefined because the variable is declared after it is used:
+
+        console.log(myVar); // undefined
+        var myVar = "Hello, world!"
+
+In this example, the myVar variable is declared after it is used, but it is hoisted to the top of the scope so the code still works. However, the variable is undefined until it is assigned a value.
+
+Source:
+
+MDN Web Docs: Hoisting - https://developer.mozilla.org/en-US/docs/Glossary/Hoisting
+
+------
+
+
 ## Looking Ahead: Terms for Next Week
 
```

Comment:
Celebrating the opportunity you are giving yourself with the efforts you placed into this assessment.